### ISSUE TITLE
CA1063: Tests fail after correcting broken code

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#1011]
About 20 tests in https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/ImplementIDisposableCorrectlyTests.cs contain testcode like:

```csharp
protected virtual Dispose(bool disposing)
{
}
```

Note the `bool` return value in this signature is missing.

This is probably a copy/paste error, because the test names do not indicate anything about handling broken code. But when I correct the signatures, tests start returning a different number of diagnostics. So the fix is not straightforward.

**Comment by bkoelman** [comment]
@duracellko Looking at commit history, you seem to know about this. Would you please take a look?

Some background: I'm working on adding a flag to the `VerifyXXX` methods that by default fail in case the test code does not compile. It's not clear to me how to resolve these.

**Comment by duracellko** [comment]
Hi @bkoelman, sorry for the broken tests.
Actually the return type of Dispose method should be void, not bool. Then the tests should be ok.
See https://github.com/duracellko/roslyn-analyzers/blob/ImplementIDisposableCorrectly/src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/ImplementIDisposableCorrectlyTests.cs

Shall I create a pull request for this? Or will you simply include it in your update?

**Comment by bkoelman** [comment]
I tried that, but then some tests fail. It would be great if you can create
a PR to fix them.

**Comment by bkoelman** [comment]
@duracellko Sorry I was too quick with my thumbs up. After I merged in your PR, I still get some errors:

```
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(663,13): Test0.cs(17,15): error CS1520: Method must have a return type
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(753,13): Test0.cs(17,52): error CS1002: ; expected
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1272,13): Test0.vb(19,16): error BC30149: Class 'C' must implement 'Property Test As Integer' for interface 'ITest'.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1272,13): Test0.vb(33,41): error BC30695: sub 'Dispose' must be declared 'Shadows' because another member with this name is declared 'Shadows'.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1316,13): Test0.vb(19,16): error BC30149: Class 'C' must implement 'Property Test As Integer' for interface 'ITest'.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1347,13): Test0.vb(19,16): error BC30149: Class 'C' must implement 'Property Test As Integer' for interface 'ITest'.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1461,13): Test0.vb(23,31): error BC31409: sub 'Dispose' must be declared 'Overloads' because another 'Dispose' is declared 'Overloads' or 'Overrides'.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1494,13): Test0.vb(23,31): error BC31409: sub 'Dispose' must be declared 'Overloads' because another 'Dispose' is declared 'Overloads' or 'Overrides'.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1918,13): Test0.vb(11,16): error BC31409: sub 'Dispose' must be declared 'Overloads' because another 'Dispose' is declared 'Overloads' or 'Overrides'.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1918,13): Test0.vb(21,44): error BC30284: sub 'Dispose' cannot be declared 'Overrides' because it does not override a sub in a base class.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1949,13): Test0.vb(11,16): error BC31409: sub 'Dispose' must be declared 'Overloads' because another 'Dispose' is declared 'Overloads' or 'Overrides'.
roslyn-analyzers\src\Microsoft.ApiDesignGuidelines.Analyzers\UnitTests\ImplementIDisposableCorrectlyTests.cs(1949,13): Test0.vb(21,29): error BC30284: sub 'Dispose' cannot be declared 'Overrides' because it does not override a sub in a base class.
```
